hikari/impl/event_manager_base.py

```diff
@@ -71,51 +71,22 @@ class EventManagerBase(event_manager.EventManager):
     is the raw event name being dispatched in lower-case.
     """
 
-    __slots__: typing.Sequence[str] = ("_app", "_consumers", "_intents", "_listeners", "_waiters")
+    __slots__: typing.Sequence[str] = ("_app", "_listeners", "_waiters")
 
     def __init__(self, app: traits.BotAware) -> None:
         self._app = app
-        self._consumers: typing.Dict[str, typing.List[ConsumerT]] = {}
-        self._intents = app.intents
         self._listeners: ListenerMapT[base_events.Event] = {}
         self._waiters: WaiterMapT[base_events.Event] = {}
 
-        for name, member in inspect.getmembers(self):
-            if name.startswith("on_"):
-                self.add_raw_consumer(name[3:], member)
-
-    def add_raw_consumer(self, name: str, consumer: ConsumerT, /) -> None:
-        name = name.casefold()
-        if name not in self._consumers:
-            self._consumers[name] = []
-
-        self._consumers[name].append(consumer)
-
-    def get_raw_consumers(self, name: str, /) -> typing.Sequence[ConsumerT]:
-        if consumers := self._consumers.get(name.casefold()):
-            return consumers.copy()
-
-        return []
-
-    def remove_raw_consumer(self, name: str, consumer: ConsumerT, /) -> None:
-        name = name.casefold()
-
-        try:
-            self._consumers[name].remove(consumer)
-
-        except ValueError:
-            raise LookupError(name, consumer) from None
-
-        if not self._consumers[name]:
-            del self._consumers[name]
-
     def consume_raw_event(
-        self, shard: gateway_shard.GatewayShard, event_name: str, payload: data_binding.JSONObject
+        self, event_name: str, shard: gateway_shard.GatewayShard, payload: data_binding.JSONObject
     ) -> None:
-        callbacks = self._consumers[event_name.casefold()]
-        asyncio.gather(
-            *(self._handle_dispatch(callback, shard, payload) for callback in callbacks), return_exceptions=True
-        )
+        try:
+            callback = getattr(self, "on_" + event_name.lower())
```

Comment:
random point but loading these into a dict instead of calling getattr every dispatch was a small optimisation, not just a part of the multi-consumer

Reply:
Ik, I was thinking that it might be better this way as it allows to add more programmatically. But after thinking about it, im going to add the consumers dict back